fastpool/src/bounded.rs

```diff
@@ -195,6 +195,51 @@ impl<M: ManageObject> Pool<M> {
         })
     }
 
+    /// Replenishes the pool with at most `most` number of new objects.
+    ///
+    /// * Returns `Ok(n)` if successfully replenished `n` objects.
+    /// * Returns `Ok(0)` if no replenishment is needed.
+    /// * Returns `Err(e)` if an error occurred while replenishing.
+    pub async fn replenish(&self, mut most: usize) -> Result<usize, M::Error> {
+        let permit = loop {
+            if most == 0 {
+                return Ok(0);
+            }
+
+            match self.permits.try_acquire(most) {
+                Some(permit) => break permit,
+                None => {
+                    most = most.min(self.permits.available_permits());
+                    continue;
+                }
+            }
+        };
+
+        most = most.saturating_sub(self.slots.lock().deque.len());
+        if most == 0 {
+            return Ok(0);
+        }
+
+        let mut idles = vec![];
+        for _ in 0..most {
+            let object = self.manager.create().await?;
```

Comment:
Considering ignore connection that failed to create. For those successfully created, still push back to the pool.